Modules/LearningSequence/classes/class.ilLearningSequenceExporter.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @author Daniel Weise <daniel.weise@concepts-and-training.de>
+ */
+class ilLearningSequenceExporter extends ilXmlExporter
+{
+	public function init()
+	{
+		global $DIC;
+
+		$this->ilias = $DIC["ilias"];
+	}
+
+	public function getXmlRepresentation($entity, $target_release, $obj_id)
+	{
+		$writer = $this->getWriter((int)$obj_id);
+		$writer->start();
+
+		return $writer->getXml();
+	}
+
+	protected function getWriter(int $obj_id): ilLearningSequenceXMLWriter
+	{
+		if ($type = ilObject::_lookupType($obj_id) != "lso") {
+			throw new Exception("Wrong type ".$type." for lso export.");
+		}
+
+		$ls_ref_id = end(ilObject::_getAllReferences($obj_id));
+		$ls_object = ilObjectFactory::getInstanceByRefId($ls_ref_id, false);
+
+		return new ilLearningSequenceXMLWriter(
+			$ls_object,
+			$this->ilias
+		);
+	}
+
+	public function getValidSchemaVersions($entity)
+	{
+		return array (
+			"5.4.0" => array(
```

Comment:
Wird das sonst auch statisch zurückgegeben? Oder kann man diese Informationen auch irgendwie von ILIAS erlangen?

Reply:
wird in allen fällen die ich mir angeschaut habe so erstellt.